チャット画面の実装

tossy9 opened the pull request:
# What
チャット画面の実装

# Why
Railsアプリケーション作成のための雛形が必要なため

Commit:
create pull request

Commit:
generate messages controller

generate messages controller and index action

command line is
$ rails g controller Messages index

Commit:
delete comment

Commit:
change root path

set root path to message’s index action

Commit:
create sass files

change application.css to application.scss
create _reset.scss, and _variable.scss

Commit:
add reset css using YUI 3 template

Commit:
add gem `font-awesome-rails`

add gem `font-awesome-rails` and bundle install

Commit:
import `font-awesome`

import `font-awesome` to application.scss file

tossy9 committed:
add Rails.gitignore

tossy9 committed:
add gitignore .sass-cache

tossy9 committed:
create top view

tossy9 committed:
add a link tag to icons

- use link_to tag
- add temporary link as root path
- icons are edit, cog, image

tossy9 committed:
modify message input form

add flex property
delete float property and min-width

tossy9 committed:
modify message form

add input forms. their types are text, file and submit.
the file select form is implemented by image icon.

tossy9 commented:
<img width="1440" alt="chat-space-view_0" src="https://user-images.githubusercontent.com/37260012/41498768-237fbc58-71b0-11e8-85c9-aecf7e4ae72a.png">

tossy9 commented:
<img width="1440" alt="2018-06-16 21 57 37" src="https://user-images.githubusercontent.com/37260012/41498779-5d3b4cf0-71b0-11e8-8a2b-b3ceb7e35fc4.png">

MrUrakou commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+        tossy
+      %ul.header__btns
+        %li
+          = link_to root_path do
```
root_pathではないので、仮置きなら"#"の方が一般的ですね。
以下同様です。

tossy9 replied:
仮置きの記載、"#"に変更します

MrUrakou commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+      %h4.messages__body
+        寝ますzzz
+    .form
+      %form.form__contents{:controller => "messages", :action => "/", :method => "post"}
```
ハッシュロケットは古い記法なのでシンボルに修正したいですね。
以下同様です。

tossy9 replied:
そうですね、シンボル記述に変更します

MrUrakou commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,207 @@
+  background-color: $font-white;
+  color: $font-blue;
+  border: 1px solid $font-blue;
+}
```
すごくいいと思います😀ただmixinはmixinsとか別フォルダを作って、mixin.scssとかファイルに記述してapplication.scssでimportしたいですね。

tossy9 replied:
別ファイルにしてimportするようにします

MrUrakou commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,66 @@
+      .header__contents.header__btn.btn
+        = link_to "Edit", root_path
+    .messages
+      %h2.messages__user
```
messageはdivタグで囲んであげたいです。
あと全部%hとするのは違和感あるので修正したいです。

tossy9 replied:
.messageで囲って.messages__*のclass名も変更します
%hは違和感なのですね。divタグで実装するよう検討します。

MrUrakou commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,207 @@
+    clear: both;
+    display: block;
+  }
+}
```
こちらも_clearfix.scssに切り出してimportしたいです。

tossy9 replied:
importするようにします

MrUrakou reviewed (commented):
すごく仕事が丁寧な印象を受けました😀
viewをどのように構造化するかはある程度お任せしますが少なくともmessageやgroupを繰り返し表示させる箇所はpartialに切り出してrenderで呼び出したいです。
サイドバーとメインのコンテンツも切り分けても良いと思います。

tossy9 committed:
change temporary path

“root_path” → “#”

tossy9 committed:
modify hash description

change traditional method to new one

tossy9 committed:
add message class

- add message class
- add flex-box mixin
- apply flex-box mixin to message and form
- delete %h tags

tossy9 committed:
add _clearfix.scss

add _clearfix.scss and import it in application.scss
delete clearfix description from messages.scss

tossy9 committed:
make messages.css partial

mkdir modules
move some import files to modules directory

tossy9 committed:
generate partial files for index.html.haml

generate 4 files
  sidebar, chat, group, message

ShuheiTsuji commented on app/views/messages/_chat.html.haml:
```diff
@@ -0,0 +1,17 @@
+  .header__contents.header__btn.btn
+    = link_to "Edit", "#"
+.messages
+  = render 'messages/message', collection: @messages
```
こちらcollection削除しても、画面表示されないでしょうか！

ShuheiTsuji commented on app/views/messages/_group.html.haml:
```diff
@@ -0,0 +1,15 @@
+    ラララ
+%ul.groups__group
+  %li.groups__name
+    安室奈美恵
```
![9775d2f88086a3b1e946b322c55b06ec](https://user-images.githubusercontent.com/16604920/42732814-76db605a-8862-11e8-91bd-66ba0d3c08be.jpg)

ShuheiTsuji replied:
LGTMです 🎉

tossy9 replied:
安室ちゃん！ありがとうございます！
